Agent crashes when assigned work with server 17.4

*  development server compiled from source at 17.4.0 tag.
 * golang agent compiled from master on Golang 1.7.5

Right after starting agent, first suspicious message is
```
<-- acknowledge
ignore acknowledge with id: , expected: cd0542f5-99f5-4e07-be4d-99042efa0552
```

Then when I scheduled a pipeline agent crashed.

## Full log

```
$ ./gocd-golang-agent_linux_amd64 
>>>>>>> go >>>>>>>
working directory: /home/tomzo/code/golang/output
fetching Go server[localhost:8154] CA certificate
fetching agent key and certificates from: https://localhost:8154/go/admin/agent
connect to: wss://localhost:8154/go/agent-websocket
--> ping
<-- acknowledge
ignore acknowledge with id: , expected: cd0542f5-99f5-4e07-be4d-99042efa0552
<-- assignWork
wait for message acknowledge timeout, id: cd0542f5-99f5-4e07-be4d-99042efa0552
--> ping
panic: Unknown message action: &{Action:assignWork Data:rO0ABXNyACljb20udGhvdWdodHdvcmtzLmdvLnJlbW90ZS53b3JrLkJ1aWxkV29yawGImRgHQhY4AgABTAAKYXNzaWdubWVudHQAMUxjb20vdGhvdWdodHdvcmtzL2dvL3JlbW90ZS93b3JrL0J1aWxkQXNzaWdubWVudDt4cHNyAC9jb20udGhvdWdodHdvcmtzLmdvLnJlbW90ZS53b3JrLkJ1aWxkQXNzaWdubWVudO3233KdF5PuAgAGTAAIYXBwcm92ZXJ0ABJMamF2YS9sYW5nL1N0cmluZztMABVidWlsZFdvcmtpbmdEaXJlY3Rvcnl0AA5MamF2YS9pby9GaWxlO0wACGJ1aWxkZXJzdAAQTGphdmEvdXRpbC9MaXN0O0wADmluaXRpYWxDb250ZXh0dAA9TGNvbS90aG91Z2h0d29ya3MvZ28vdXRpbC9jb21tYW5kL0Vudmlyb25tZW50VmFyaWFibGVDb250ZXh0O0wAEW1hdGVyaWFsUmV2aXNpb25zdAAuTGNvbS90aG91Z2h0d29ya3MvZ28vZG9tYWluL01hdGVyaWFsUmV2aXNpb25zO0wABHBsYW50ACRMY29tL3Rob3VnaHR3b3Jrcy9nby9kb21haW4vSm9iUGxhbjt4cHQAB2NoYW5nZXNzcgAMamF2YS5pby5GaWxlBC2kRQ4N5P8DAAFMAARwYXRocQB+AAR4cHQADnBpcGVsaW5lcy90ZXN0dwIAL3hzcgATamF2YS51dGlsLkFycmF5TGlzdHiB0h2Zx2GdAwABSQAEc2l6ZXhwAAAAAXcEAAAAAXNyADFjb20udGhvdWdodHdvcmtzLmdvLmRvbWFpbi5idWlsZGVyLkNvbW1hbmRCdWlsZGVyHhcWIeukenwCAAFMAARhcmdzcQB+AAR4cgA1Y29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uYnVpbGRlci5CYXNlQ29tbWFuZEJ1aWxkZXKeBPFvfPXmPwIAA0wAB2NvbW1hbmRxAH4ABEwAC2Vycm9yU3RyaW5ncQB+AARMAAp3b3JraW5nRGlycQB+AAV4cgAqY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uYnVpbGRlci5CdWlsZGVyENRK8a8ap90CAANMAA1jYW5jZWxCdWlsZGVydAAsTGNvbS90aG91Z2h0d29ya3MvZ28vZG9tYWluL2J1aWxkZXIvQnVpbGRlcjtMAApjb25kaXRpb25zdAApTGNvbS90aG91Z2h0d29ya3MvZ28vZG9tYWluL1J1bklmQ29uZmlncztMAAtkZXNjcmlwdGlvbnEAfgAEeHBzcgA9Y29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uYnVpbGRlci5CdWlsZGVyRm9yS2lsbEFsbENoaWxkVGFza73g86ZGxONaAgAAeHEAfgATc3IALmNvbS50aG91Z2h0d29ya3MuZ28uZG9tYWluLmJ1aWxkZXIuTnVsbEJ1aWxkZXIuzPEr3Is3swIAAHhxAH4AE3BzcgAnY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uUnVuSWZDb25maWdzAmoWlK7O4n4CAAFMAAxjb25maWdFcnJvcnN0AClMY29tL3Rob3VnaHR3b3Jrcy9nby9kb21haW4vQ29uZmlnRXJyb3JzO3hyACljb20udGhvdWdodHdvcmtzLmdvLmRvbWFpbi5CYXNlQ29sbGVjdGlvblSxTLgh5n3TAgAAeHEAfgAPAAAAAHcEAAAAAHhzcgAnY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uQ29uZmlnRXJyb3Jz3u0adIZKKbwCAAB4cgARamF2YS51dGlsLkhhc2hNYXAFB9rBwxZg0QMAAkYACmxvYWRGYWN0b3JJAAl0aHJlc2hvbGR4cD9AAAAAAAAAdwgAAAAQAAAAAHh0AAROVUxMc3EAfgAbAAAAAHcEAAAAAHhzcQB+AB8/QAAAAAAAAHcIAAAAEAAAAAB4dAAVS2lsbHMgY2hpbGQgcHJvY2Vzc2Vzc3EAfgAbAAAAAHcEAAAAAHhzcQB+AB8/QAAAAAAAAHcIAAAAEAAAAAB4dAAEcmFrZXEAfgAodAAAc3EAfgAMdAAOcGlwZWxpbmVzL3Rlc3R3AgAveHEAfgApeHNyADtjb20udGhvdWdodHdvcmtzLmdvLnV0aWwuY29tbWFuZC5FbnZpcm9ubWVudFZhcmlhYmxlQ29udGV4dAPlOtNvzF3LAgABTAAKcHJvcGVydGllc3EAfgAGeHBzcQB+AA8AAAAAdwQAAAAAeHNyACxjb20udGhvdWdodHdvcmtzLmdvLmRvbWFpbi5NYXRlcmlhbFJldmlzaW9uc0IYdBN3MxFcAgABTAAJcmV2aXNpb25zcQB+AAZ4cHNxAH4ADwAAAAF3BAAAAAFzcgArY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uTWF0ZXJpYWxSZXZpc2lvbiN23MoFo2z7AgADWgAHY2hhbmdlZEwACG1hdGVyaWFsdAAvTGNvbS90aG91Z2h0d29ya3MvZ28vZG9tYWluL21hdGVyaWFscy9NYXRlcmlhbDtMAA1tb2RpZmljYXRpb25zdAA0TGNvbS90aG91Z2h0d29ya3MvZ28vZG9tYWluL21hdGVyaWFscy9Nb2RpZmljYXRpb25zO3hwAXNyADRjb20udGhvdWdodHdvcmtzLmdvLmNvbmZpZy5tYXRlcmlhbHMuZ2l0LkdpdE1hdGVyaWFstXW2UDF2HpYCAARaAAxzaGFsbG93Q2xvbmVMAAZicmFuY2hxAH4ABEwAD3N1Ym1vZHVsZUZvbGRlcnEAfgAETAADdXJsdAAuTGNvbS90aG91Z2h0d29ya3MvZ28vdXRpbC9jb21tYW5kL1VybEFyZ3VtZW50O3hyADBjb20udGhvdWdodHdvcmtzLmdvLmNvbmZpZy5tYXRlcmlhbHMuU2NtTWF0ZXJpYWwkkaBS1h1IZgIABFoACmF1dG9VcGRhdGVaAAxpbnZlcnRGaWx0ZXJMAAZmaWx0ZXJ0AC1MY29tL3Rob3VnaHR3b3Jrcy9nby9jb25maWcvbWF0ZXJpYWxzL0ZpbHRlcjtMAAZmb2xkZXJxAH4ABHhyADVjb20udGhvdWdodHdvcmtzLmdvLmNvbmZpZy5tYXRlcmlhbHMuQWJzdHJhY3RNYXRlcmlhbOj8QHY8nyuiAgAGTAAQYXR0cmlidXRlc0ZvclhtbHQAD0xqYXZhL3V0aWwvTWFwO0wAC2ZpbmdlcnByaW50cQB+AARMAARuYW1ldAAyTGNvbS90aG91Z2h0d29ya3MvZ28vY29uZmlnL0Nhc2VJbnNlbnNpdGl2ZVN0cmluZztMABlwaXBlbGluZVVuaXF1ZUZpbmdlcnByaW50cQB+AARMAAtzcWxDcml0ZXJpYXEAfgA7TAAEdHlwZXEAfgAEeHIAK2NvbS50aG91Z2h0d29ya3MuZ28uZG9tYWluLlBlcnNpc3RlbnRPYmplY3S8ScmMkvThSAIAAUoAAmlkeHAAAAAAAAAAA3B0AEAzNDEyNzljNzM1NDdiODNkNzJlNzljNWJjOTlkOWFiYTA0MmU3ODljMzRlZjlmZDkyMjA1OWFjODhlZmUyZDMzcHBzcgAlamF2YS51dGlsLkNvbGxlY3Rpb25zJFVubW9kaWZpYWJsZU1hcPGlqP509QdCAgABTAABbXEAfgA7eHBzcgAXamF2YS51dGlsLkxpbmtlZEhhc2hNYXA0wE5cEGzA+wIAAVoAC2FjY2Vzc09yZGVyeHEAfgAgP0AAAAAAAAx3CAAAABAAAAADdAAEdHlwZXQAC0dpdE1hdGVyaWFsdAADdXJsdAAxaHR0cHM6Ly9naXRodWIuY29tL2dvY2QtY29udHJpYi9nb2NkLWdvbGFuZy1hZ2VudHQABmJyYW5jaHQABm1hc3RlcngAcQB+AEUBAHBwAHEAfgBJcHNyACxjb20udGhvdWdodHdvcmtzLmdvLnV0aWwuY29tbWFuZC5VcmxBcmd1bWVudJ1JVm0hUa3XAgABTAADdXJscQB+AAR4cgAwY29tLnRob3VnaHR3b3Jrcy5nby51dGlsLmNvbW1hbmQuQ29tbWFuZEFyZ3VtZW505NxH9S0tOdUCAAB4cHEAfgBHc3IAMmNvbS50aG91Z2h0d29ya3MuZ28uZG9tYWluLm1hdGVyaWFscy5Nb2RpZmljYXRpb25zWqexhwVbNF0CAAB4cQB+AB0AAAABdwQAAAABc3IAMWNvbS50aG91Z2h0d29ya3MuZ28uZG9tYWluLm1hdGVyaWFscy5Nb2RpZmljYXRpb25UsLUQCkCfUAMAC0wADmFkZGl0aW9uYWxEYXRhcQB+AARMABFhZGRpdGlvbmFsRGF0YU1hcHQAE0xqYXZhL3V0aWwvSGFzaE1hcDtMAAdjb21tZW50cQB+AARMAAxlbWFpbEFkZHJlc3NxAH4ABEwABWZpbGVzdAAPTGphdmEvdXRpbC9TZXQ7TAAQbWF0ZXJpYWxJbnN0YW5jZXQALUxjb20vdGhvdWdodHdvcmtzL2dvL2RvbWFpbi9NYXRlcmlhbEluc3RhbmNlO0wADG1vZGlmaWVkVGltZXQAEExqYXZhL3V0aWwvRGF0ZTtMAApwaXBlbGluZUlkdAAQTGphdmEvbGFuZy9Mb25nO0wADXBpcGVsaW5lTGFiZWxxAH4ABEwACHJldmlzaW9ucQB+AARMAAh1c2VyTmFtZXEAfgAEeHEAfgA9AAAAAAAAAAF0ACJCYXJyb3cgS3dhbiA8YmFycm93a3dhbkB5YWhvby5jb20+dABjTWVyZ2UgcHVsbCByZXF1ZXN0ICM0MyBmcm9tIGJhcnJvd2t3YW4vbWFzdGVyCgojNDIgV2Vic29ja2V0IG1lc3NhZ2UgJ2FjaycgY2hhbmdlZCB0byAnYWNrbm93bGVkZ2UncHQAKGFmNjAwMWZjM2Q4OWEzZmRmMWZmNTlkODhkNjAwODAxMDZmZDM5ZDRwcHNyABJqYXZhLnNxbC5UaW1lc3RhbXAmGNXIAVO/ZQIAAUkABW5hbm9zeHIADmphdmEudXRpbC5EYXRlaGqBAUtZdBkDAAB4cHcIAAABWvmDAkh4AAAAAHBzcgA8Y29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4ubWF0ZXJpYWxzLmdpdC5HaXRNYXRlcmlhbEluc3RhbmNlZHpQ8yIVxDQCAAB4cgArY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uTWF0ZXJpYWxJbnN0YW5jZZiN1cRegfmhAgARTAAOYWRkaXRpb25hbERhdGFxAH4ABEwAEWFkZGl0aW9uYWxEYXRhTWFwcQB+ADtMAAZicmFuY2hxAH4ABEwADmNoZWNrRXh0ZXJuYWxzdAATTGphdmEvbGFuZy9Cb29sZWFuO0wADWNvbmZpZ3VyYXRpb25xAH4ABEwABmRvbWFpbnEAfgAETAALZmluZ2VycHJpbnRxAH4ABEwADWZseXdlaWdodE5hbWVxAH4ABEwADHBpcGVsaW5lTmFtZXEAfgAETAALcHJvamVjdFBhdGhxAH4ABEwACXN0YWdlTmFtZXEAfgAETAAPc3VibW9kdWxlRm9sZGVycQB+AARMAAN1cmxxAH4ABEwACnVzZVRpY2tldHNxAH4AXkwACHVzZXJuYW1lcQB+AARMAAR2aWV3cQB+AARMAAl3b3Jrc3BhY2VxAH4ABHhxAH4APQAAAAAAAAADcHB0AAZtYXN0ZXJwcHBxAH4AP3QAJGIzNmQxYWIxLWYzZTctNGI2Ni1hNmIyLTYwZjQwN2Y4MTkxZnBwcHB0ADFodHRwczovL2dpdGh1Yi5jb20vZ29jZC1jb250cmliL2dvY2QtZ29sYW5nLWFnZW50cHBwcHNyABdqYXZhLnV0aWwuTGlua2VkSGFzaFNldNhs11qV3SoeAgAAeHIAEWphdmEudXRpbC5IYXNoU2V0ukSFlZa4tzQDAAB4cHcMAAAAED9AAAAAAAAAeHh4eHNyACljb20udGhvdWdodHdvcmtzLmdvLmRvbWFpbi5EZWZhdWx0Sm9iUGxhbpZ+I1Cicbg2AgAMWgAPY2xlYW5Xb3JraW5nRGlyWgAOZmV0Y2hNYXRlcmlhbHNKAAVqb2JJZEoACnBpcGVsaW5lSWRMAAlhZ2VudFV1aWRxAH4ABEwADmVsYXN0aWNQcm9maWxldAAzTGNvbS90aG91Z2h0d29ya3MvZ28vY29uZmlnL2VsYXN0aWMvRWxhc3RpY1Byb2ZpbGU7TAAKZ2VuZXJhdG9yc3QAOUxjb20vdGhvdWdodHdvcmtzL2dvL2NvbmZpZy9BcnRpZmFjdFByb3BlcnRpZXNHZW5lcmF0b3JzO0wACmlkZW50aWZpZXJ0ACpMY29tL3Rob3VnaHR3b3Jrcy9nby9kb21haW4vSm9iSWRlbnRpZmllcjtMAAVwbGFuc3QAKkxjb20vdGhvdWdodHdvcmtzL2dvL2NvbmZpZy9BcnRpZmFjdFBsYW5zO0wACXJlc291cmNlc3QAJkxjb20vdGhvdWdodHdvcmtzL2dvL2NvbmZpZy9SZXNvdXJjZXM7TAAQdHJpZ2dlclZhcmlhYmxlc3QAN0xjb20vdGhvdWdodHdvcmtzL2dvL2NvbmZpZy9FbnZpcm9ubWVudFZhcmlhYmxlc0NvbmZpZztMAAl2YXJpYWJsZXNxAH4AbHhwAAEAAAAAAAAAAQAAAAAAAAABcHBzcgA3Y29tLnRob3VnaHR3b3Jrcy5nby5jb25maWcuQXJ0aWZhY3RQcm9wZXJ0aWVzR2VuZXJhdG9yc5PWECAdi6W/AgABTAAMY29uZmlnRXJyb3JzcQB+ABx4cQB+AB0AAAAAdwQAAAAAeHNxAH4AHz9AAAAAAAAAdwgAAAAQAAAAAHhzcgAoY29tLnRob3VnaHR3b3Jrcy5nby5kb21haW4uSm9iSWRlbnRpZmllcm76zQ7zyTpbAgAITAAHYnVpbGRJZHEAfgBUTAAJYnVpbGROYW1lcQB+AARMAA9waXBlbGluZUNvdW50ZXJ0ABNMamF2YS9sYW5nL0ludGVnZXI7TAANcGlwZWxpbmVMYWJlbHEAfgAETAAMcGlwZWxpbmVOYW1lcQB+AARMAA5yZXJ1bk9mQ291bnRlcnEAfgByTAAMc3RhZ2VDb3VudGVycQB+AARMAAlzdGFnZU5hbWVxAH4ABHhwc3IADmphdmEubGFuZy5Mb25nO4vkkMyPI98CAAFKAAV2YWx1ZXhyABBqYXZhLmxhbmcuTnVtYmVyhqyVHQuU4IsCAAB4cAAAAAAAAAABdAAKZGVmYXVsdEpvYnNyABFqYXZhLmxhbmcuSW50ZWdlchLioKT3gYc4AgABSQAFdmFsdWV4cQB+AHUAAAABdAABMXQABHRlc3RwdAABMXQADGRlZmF1bHRTdGFnZXNyAChjb20udGhvdWdodHdvcmtzLmdvLmNvbmZpZy5BcnRpZmFjdFBsYW5zDjZ9NVvFi0YCAAFMAAxjb25maWdFcnJvcnNxAH4AHHhxAH4AHQAAAAB3BAAAAAB4c3EAfgAfP0AAAAAAAAB3CAAAABAAAAAAeHNyACRjb20udGhvdWdodHdvcmtzLmdvLmNvbmZpZy5SZXNvdXJjZXNmC4Esy3CWugIAAUwADGNvbmZpZ0Vycm9yc3EAfgAceHEAfgAdAAAAAHcEAAAAAHhzcQB+AB8/QAAAAAAAAHcIAAAAEAAAAAB4c3IANWNvbS50aG91Z2h0d29ya3MuZ28uY29uZmlnLkVudmlyb25tZW50VmFyaWFibGVzQ29uZmlnfw//JWB8xK8CAAFMAAxjb25maWdFcnJvcnNxAH4AHHhxAH4AHQAAAAB3BAAAAAB4c3EAfgAfP0AAAAAAAAB3CAAAABAAAAAAeHNxAH4AhAAAAAB3BAAAAAB4c3EAfgAfP0AAAAAAAAB3CAAAABAAAAAAeA== AcknowledgeId:}

goroutine 1 [running]:
panic(0x712180, 0xc42027b7a0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/gocd-contrib/gocd-golang-agent/agent.processMessage(0xc420286cc0, 0xc42020cd50, 0xc420202cc0, 0xc42014bd7f, 0xc4202a3001)
	/ide/work/src/github.com/gocd-contrib/gocd-golang-agent/agent/agent.go:146 +0x1c5
github.com/gocd-contrib/gocd-golang-agent/agent.Start(0x0, 0x0)
	/ide/work/src/github.com/gocd-contrib/gocd-golang-agent/agent/agent.go:102 +0x28b
main.main()
	/ide/work/src/github.com/gocd-contrib/gocd-golang-agent/main.go:44 +0xab
```

--- a/agent/registration.go
+++ b/agent/registration.go
@@ -144,6 +144,7 @@ func registerData() map[string]string {
 		"agentAutoRegisterHostname":     config.Hostname,
 		"elasticAgentId":                config.AgentAutoRegisterElasticAgentId,
 		"elasticPluginId":               config.AgentAutoRegisterElasticPluginId,
+		"supportsBuildCommandProtocol":  "true",
 	}
 }
 

